### DISCUSSION TITLE
Nix feature toggle not working on Mend Renovate bot

### DISCUSSION THREAD

**Discussion by hbjydev** [renovatebot/renovate#29263] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

I have this config in `.github/renovate.json`:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
    ":semanticCommitTypeAll(chore)"
  ],
  "lockFileMaintenance": {
    "enabled": true,
    "extends": [
      "schedule:weekly"
    ]
  },
  "nix": {
    "enabled": true
  }
}
```

The base config and the semantic commit extension both work fine, but Renovate doesn't seem to be detecting the Nix flake in that repo. There is a `flake.nix` and `flake.lock` in there, but in the Issue Dashboard (ALT-F4-LLC/vms.nix#16), it's not even being listed as a dependency source.

### Logs (if relevant)

https://gist.github.com/hbjydev/ee5766e3445178fb798c35a704b5d165

I think that during a _run_, it's finding the flake (line 573 of the log gist above)...

```
DEBUG: Matched 1 file(s) for manager nix: flake.nix
```

**Comment by rarkins** [comment]
The log indicates that nix is being enabled. But please read the docs and open feature requests to better understand its limitations

**Reply by hbjydev** [reply]
I have read through, I'm not sure what limitations you mean?

**Reply by hbjydev** [reply]
If you're referring to https://github.com/renovatebot/renovate/discussions/25514 this isn't my issue, my issue is that it's not actually doing anything with the flake _at all_.

**Reply by rarkins** [reply]
What are you expecting it to do specifically? Eg give an example PR change you'd expect

**Reply by hbjydev** [reply]
I'd expect that on the schedule I provide, it raises a PR with an updated `flake.lock`, but again, the issue is that it _finds_ the flake, but doesn't pick it up as a dependency source, as shown in the dependency dashboard.

**Reply by rarkins** [reply]
With some modifications I was able to get https://github.com/renovate-reproductions/29263/pull/1/files

The problem appears to be that Renovate finds no dependencies, so it returns no package files. With no package files, there's no lock file maintenance to do.

I think we can modify this to return the package file _if_ there's also a lock file next to it.

**Reply by rarkins** [reply, marked as answer]
I've created an issue for this: https://github.com/renovatebot/renovate/issues/29380

I have the working code stashed locally and will try to push it next week

**Reply by hbjydev** [reply]
Thanks @rarkins :)
